File: src/Aresta.ts
import Vertice from "./Vertice";

export default class Aresta{
    private origem: Vertice;
    private destino: Vertice;
    private peso: number;

    public constructor(origem: Vertice, destino: Vertice, peso?: number){
        this.origem = origem;
        this.destino = destino;
        this.peso = 0;
    }

    get getOrigem(): Vertice{
        return this.origem;
    }

    set setOrigem(origem: Vertice){
        this.origem = origem;
    }

    get getDestino(): Vertice{
        return this.destino;
    }

    set setDestino(destino: Vertice){
        this.destino = destino;
    }

    get getPeso(): number{
        return this.peso;
    }

    set setPeso(peso: number){
        this.peso = peso;
    }

    idDestino(){
        return this.destino.getId;
    }

}

File: Naive/src/Aresta.java
public class Aresta {
     public Vertice origem;
    public Vertice destino;



    private double peso;


    public Aresta(Vertice origem, Vertice destino,double peso) {
        this.origem = origem;
        this.destino = destino;
        this.peso =peso;
    }

    public Vertice getOrigem() {
        return this.origem;
    }

    public void setOrigem(Vertice origem) {
        this.origem = origem;
    }

    public Vertice getDestino() {
        return this.destino;
    }

    public void setDestino(Vertice destino) {
        this.destino = destino;
    }
    public double getPeso() {
        return peso;
    }
    public Vertice destino() {
        return this.destino;
    }

    public void setPeso(double peso) {
        this.peso = peso;
    }



}
